lib/common/xml.c

```diff
@@ -2647,11 +2647,19 @@ pcmk__xml_artefact_path(enum pcmk__xml_artefact_ns ns, const char *filespec)
     switch (ns) {
         case pcmk__xml_artefact_ns_legacy_rng:
         case pcmk__xml_artefact_ns_base_rng:
-            ret = crm_strdup_printf("%s/%s.rng", base, filespec);
+            if (g_str_has_suffix(filespec, ".rng")) {
```

Comment:
Huh, I wonder why I created `pcmk__ends_with()`. Maybe I didn't realize this existed, or maybe this doesn't handle NULLs.

Reply:
Dunno, but switched to using that.  We've got this stuff, might as well use it.